### ISSUE TITLE
why the Graph output_names =['net/fc8/BiasAdd']?

### ISSUE THREAD

**Issue by zhaohu19870717** [dkorobchenko-nv/tensorrt-demo#2]
first i think you did a very great job,i am appreciate for it,however  when i try to run the demo, i meet some problems like this:  InvalidArgumentError (see above for traceback): LossTensor is inf or nan : Tensor had NaN values  

[[Node: train_op/CheckNumerics = CheckNumerics[T=DT_FLOAT, _class=["loc:@train_op/control_dependency"], message="LossTensor is inf or nan", _device="/job:localhost/replica:0/task:0/device:GPU:0"](loss/sparse_softmax_cross_entropy_loss/value, ^net/conv5_2/BatchNorm/AssignMovingAvg_1, ^net/conv1_1/BatchNorm/AssignMovingAvg, ^net/conv1_1/BatchNorm/AssignMovingAvg_1, ^net/conv2_1/BatchNorm/AssignMovingAvg, ^net/conv2_1/BatchNorm/AssignMovingAvg_1, ^net/conv3_1/BatchNorm/AssignMovingAvg, ^net/conv3_1/BatchNorm/AssignMovingAvg_1, ^net/conv3_2/BatchNorm/AssignMovingAvg, ^net/conv3_2/BatchNorm/AssignMovingAvg_1, ^net/conv4_1/BatchNorm/AssignMovingAvg, ^net/conv4_1/BatchNorm/AssignMovingAvg_1, ^net/conv4_2/BatchNorm/AssignMovingAvg, ^net/conv4_2/BatchNorm/AssignMovingAvg_1, ^net/conv5_1/BatchNorm/AssignMovingAvg, ^net/conv5_1/BatchNorm/AssignMovingAvg_1, ^net/conv5_2/BatchNorm/AssignMovingAvg)]]
         [[Node: train_op/control_dependency/_421 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_699_train_op/control_dependency", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
 i know there might be some reason for it , i don't know what the reason is,so maybe you can help me ,thank you very much ,what's more,why the output node name is net/fc8/BiasAdd, i think the output node name might be conv5_2/BatchNorm/AssigMovingAvg ,plz tell me which one is the right answer and why,thanks

**Comment by dkorobchenko** [comment]
Hi,
1) In case if you are using your own dataset or modified version of the network, please try to reproduce the training procedure with the ImageNet dataset and the original network form the repository
2) If (1) is not the case, please try to use the updated version of the code and use docker containers (included in the repo) to match versions of all the frameworks and libraries. Please check updated README
3) The "output node name" is the name of the output of the last layer. Our last layer is Fully-connected layer "fc8". The layer has Matrix-Multiplication operation and Bias-Add operation. Adding the bias is the final step to produce output logits (before softmax). You can check details in TensorBoard graph visualization
